lib/widgets/code_block.dart

```diff
@@ -1,267 +1,549 @@
 
-// .k { color: hsl(332deg 70% 38%); }
-final _kCodeBlockStyleK = TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor());
+      // .k { color: hsl(332deg 70% 38%); }
+      k: TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor()),
 
-// .o { color: hsl(332deg 70% 38%); }
-final _kCodeBlockStyleO = TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor());
+      // .o { color: hsl(332deg 70% 38%); }
+      o: TextStyle(color: const HSLColor.fromAHSL(1, 332, 0.7, 0.38).toColor()),
 
-// .cm { color: hsl(180deg 33% 37%); font-style: italic; }
-final _kCodeBlockStyleCm = TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic);
+      // .cm { color: hsl(180deg 33% 37%); font-style: italic; }
+      cm: TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic),
 
-// .cp { color: hsl(38deg 100% 36%); }
-final _kCodeBlockStyleCp = TextStyle(color: const HSLColor.fromAHSL(1, 38, 1, 0.36).toColor());
+      // .cp { color: hsl(38deg 100% 36%); }
+      cp: TextStyle(color: const HSLColor.fromAHSL(1, 38, 1, 0.36).toColor()),
 
-// .c1 { color: hsl(0deg 0% 67%); font-style: italic; }
-final _kCodeBlockStyleC1 = TextStyle(color: const HSLColor.fromAHSL(1, 0, 0, 0.67).toColor(), fontStyle: FontStyle.italic);
+      // .c1 { color: hsl(0deg 0% 67%); font-style: italic; }
+      c1: TextStyle(color: const HSLColor.fromAHSL(1, 0, 0, 0.67).toColor(), fontStyle: FontStyle.italic),
 
-// .cs { color: hsl(180deg 33% 37%); font-style: italic; }
-final _kCodeBlockStyleCs = TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic);
+      // .cs { color: hsl(180deg 33% 37%); font-style: italic; }
+      cs: TextStyle(color: const HSLColor.fromAHSL(1, 180, 0.33, 0.37).toColor(), fontStyle: FontStyle.italic),
 
-// .gd { color: hsl(0deg 100% 31%); }
-final _kCodeBlockStyleGd = TextStyle(color: const HSLColor.fromAHSL(1, 0, 1, 0.31).toColor());
+      // .gd { color: hsl(0deg 100% 31%); }
+      gd: TextStyle(color: const HSLColor.fromAHSL(1, 0, 1, 0.31).toColor()),
 
-// .ge { font-style: italic; }
-const _kCodeBlockStyleGe = TextStyle(fontStyle: FontStyle.italic);
+      // .ge { font-style: italic; }
+      ge: const TextStyle(fontStyle: FontStyle.italic),
 
-// .gr { color: hsl(0deg 100% 50%); }
-final _kCodeBlockStyleGr = TextStyle(color: const HSLColor.fromAHSL(1, 0, 1, 0.50).toColor());
+      // .gr { color: hsl(0deg 100% 50%); }
+      gr: TextStyle(color: const HSLColor.fromAHSL(1, 0, 1, 0.50).toColor()),
 
-// .gh { color: hsl(240deg 100% 25%); font-weight: bold; }
-final _kCodeBlockStyleGh = TextStyle(color: const HSLColor.fromAHSL(1, 240, 1, 0.25).toColor(), fontWeight: FontWeight.bold);
+      // .gh { color: hsl(240deg 100% 25%); font-weight: bold; }
+      gh: TextStyle(color: const HSLColor.fromAHSL(1, 240, 1, 0.25).toColor()).merge(bold),
```

Comment:
Hmm I see. The `FontWeight.bold` was getting ignored because of the `wght` variation, which leaked through from the ambient TextStyle (which in turn came from the ContentTheme). Is that right?

It'd be good to add a few words to the commit message about the visible effect of this. I wasn't sure what to expect before I opened up the screenshots at full size and compared them.

Reply:
In particular:

    And while we're add it, add some tests that spot-check the weight of
    a non-bold span and a bold span.

Those tests aren't just a by-the-way — they're regression tests for the bug this fixes :slightly_smiling_face:

Reply:
> Hmm I see. The `FontWeight.bold` was getting ignored because of the `wght` variation, which leaked through from the ambient TextStyle (which in turn came from the ContentTheme). Is that right?

No, I don't think that's quite right. The "Before" screenshot already shows the text being bolder than surrounding text, just not in the way we want it to be. I think this is because of:

- https://github.com/flutter/flutter/issues/136779

For glyphs that render in a variable-weight font, we want to specify a `wght` value, so that they appear as the font designer intended. Just setting `TextStyle.fontWeight` by itself doesn't do that—not until https://github.com/flutter/flutter/issues/148026 anyway—and our way of setting `wght` is to use `weightVariableTextStyle`. These glyphs did _have_ a `wght` value before this change, but it was wrong. It should be 700, but it was 400, from the ambient `TextStyle` from `ContentTheme`.

Reply:
Yeah, I guess "ignored" isn't right. But as you say, it (the `FontWeight.bold`) wasn't having the desired effect.